### ISSUE TITLE
U - As the MDN PM I want the ability to limit the banner for specific groups, so we can roll this out incrementely

### ISSUE THREAD

**Issue by atopal** [mdn/kuma#6555]
# User story
As the MDN PM I want the ability to limit the banner for specific groups, so we can roll this out incrementally

# Acceptance criteria

- [x] Banner can be turned on and off by PM without a code push
- [x] Banner can be limited to logged in users
- [x] Banner can be limited to a subset of all users (eg 50% of visitors, but we can postpone this if it's hard)

# Tasks
*If relevant, you can make a checklist for tasks.*

1. - [ ] T - Subscription Banner ~ https://github.com/mdn/kuma/issues/6577

# Context
We'll show the banner only to logged in users in the first iteration, and even afterwards, we might opt to only show it to beta users for a while, and we need a way to quickly turn it off again if necessary

# Design

- Signed in user:
 https://share.goabstract.com/857f8a74-8fdd-41e6-9d78-d5c838d4054a?mode=design
- Non-signed in users (but with an account): 
 https://share.goabstract.com/857f8a74-8fdd-41e6-9d78-d5c838d4054a?collectionLayerId=1c005660-47c6-4d98-aac5-bc533fa9ae26&mode=design
- Non-signed in users (without an account): 
https://share.goabstract.com/857f8a74-8fdd-41e6-9d78-d5c838d4054a?collectionLayerId=0083a875-bdc1-49e2-9215-9a0c59a4cab6&mode=design
- And then the specific use-case where you have the banner take you to the user profile/settings page(?): https://share.goabstract.com/80aff1b8-a8da-4250-ab33-f6426d45b2e1?mode=design

**Comment by tobinmori** [comment]
task #6577

**Comment by peterbe** [comment]
I don't know what action this issue can possibly become. We already have [a Waffle Flag called `subscription` in production](https://developer.mozilla.org/admin/waffle/flag/135/change/).

**Comment by atopal** [comment]
@peterbe if I understand correctly that waffle flag toggles on an off the subscription form on the user profile. This user story is about the banner that can be shown to non-logged in users as well as logged in users.

**Comment by peterbe** [comment]
Are you advocating having two distinct waffle flags? One for enabling the banner and one for enabling the subscription form?
If a user gets a Yes on the banner she gets a link to the profile edit page (with the `#subscription` anchor on it so it scrolls down to the form). If you get a No on the contribution form, you would have sent her to a link that leads to nowhere. Right?

But yeah, I could see the benefit of a banner specific waffle flag (or switch) so that can throttle the banner. But shouldn't that then be for a subset of the users who already get the contribution form? I.e. the subscription banner's presence should be coded like this:

```
if (waffle.flags['subscription'] && waffle.flag['subscription_banner']) {
  return <SubscriptionBanner/>
} else {
  return null
}
```

The thing that my brain can't parse is *"...banner that can be shown to non-logged in users"* which would mean you might, theoretically, display a link to something that won't show up. Because the Subscription form will always require that you're logged in (independent of waffle flag).

**Comment by schalkneethling** [comment]
@peterbe So the scenarios that can play out are evolving :)

Currently, we have a form on the edit profile page and its visibility is determined by the waffle flag `subscriptions`

Once the new banner land we can have

1. The form on edit profile and no banner
2. The banner on the landing page and docs pages and the form on edit profile

Once the above two have landed in production we will move onto the next iteration which is showing the banner to all users. Part of this work is to enable a user to sign up right from the page where the banner was displayed via a modal interaction flow.

At this point, we might retire the form on the edit profile page at which point we should consolidate the flags to use just one.

Until we reach that point though, we would need two flags to control the form on edit profile independent from the banner. We should just be sure to not turn the banner on and the form off for now 😺

**Comment by tobinmori** [comment]
> Are you advocating having two distinct waffle flags? One for enabling the banner and one for enabling the subscription form?

In short, 'yes' to this @peterbe 

As @schalkneethling is saying - they are orthogonal functionalities.

**Comment by peterbe** [comment]
> @peterbe So the scenarios that can play out are evolving :)
> 
> Currently, we have a form on the edit profile page and its visibility is determined by the waffle flag `subscriptions`
> 
> Once the new banner land we can have
> 
>     1. The form on edit profile and no banner
> 
>     2. The banner on the landing page and docs pages and the form on edit profile
> 
> 
> Once the above two have landed in production we will move onto the next iteration which is showing the banner to all users. Part of this work is to enable a user to sign up right from the page where the banner was displayed via a modal interaction flow.
> 
> At this point, we might retire the form on the edit profile page at which point we should consolidate the flags to use just one.
> 
> Until we reach that point though, we would need two flags to control the form on edit profile independent from the banner. We should just be sure to not turn the banner on and the form off for now 😺

Gotcha! Then, rename it to `subscriptions_banner` (or `subscription_banner`) and this PR will need to have a DB migration that creates that waffle flag. A name like that would still make sense once the banner can trigger a modal. And who knows, we might want to have a general Subscriptions page, still enabled, for all, but perhaps no banner (in the future). 

But the banner is a *subset* of people who get the subscription form on the edit profile page, no?
If so, the check should be
```javascript
if (waffle.flags['subscription'] && waffle.flag['subscription_banner']) {
  return <SubscriptionBanner/>
} else {
  return null
}
```
However, perhaps we can just take our chances and assume that that won't be necessary. I.e. that whoever enabled the `subscription_banner` waffle flag also enabled the `subscription` waffle flag. 

Now, I think I understand why you want to do things depending on the user being signed in.

**Comment by peterbe** [comment]
The subscription banner is now implemented and even landed in `master`. The ability to hide the banner if you're an active subscription is [also landed](https://github.com/mdn/kuma/pull/6706). 
What else is needed to this user story to close this?

**Comment by schalkneethling** [comment]
@atopal Shall we close this one?

**Comment by atopal** [comment]
I'm not sure, how can I test this?

**Comment by schalkneethling** [comment]
> I'm not sure, how can I test this?

Until we can roll it out, it is tricky. It is controlled via Waffle so, if you have confidence in Waffle, this should be fine ;)

**Comment by peterbe** [comment]
@tobinmori I'm guessing you re-opened it because of the Subscription banner was still requiring to you to be authenticated which means it was at the mercy of the waffle flag *and* auth. 
https://github.com/mdn/kuma/pull/6830 is now merged into master. 
So, close this user story issue??

**Comment by tobinmori** [comment]
Done! Thanks @peterbe.
